target/linux/mediatek/dts/mt7622-netgear-wax206.dts

```diff
@@ -0,0 +1,603 @@
+		mediatek,bmt-v2;
+		nand-ecc-engine = <&snfi>;
+		reg = <0>;
+		spi-rx-bus-width = <4>;
+		spi-tx-bus-width = <4>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+	/*
+	[    1.456820] 0x000000000000-0x000000080000 : "Preloader"
+	[    1.463323] 0x000000080000-0x0000000c0000 : "ATF"
+	[    1.468865] 0x0000000c0000-0x000000140000 : "Bootloader"
+	[    1.475339] 0x000000140000-0x0000001c0000 : "Config"
+	[    1.481469] 0x0000001c0000-0x0000002c0000 : "Factory"
+	[    1.488403] 0x0000002c0000-0x0000028c0000 : "ImageA"
+	[    1.546793] 0x0000002e0000-0x0000028c0000 : "Kernel"
+	[    1.609321] 2 fit-fw partitions found on MTD device Kernel
+	[    1.614809] 0x0000002e0000-0x000000580000 : "kernel"
+	[    1.623890] 0x000000580000-0x0000028c0000 : "rootfs"
+	[    1.678471] mtd: device 8 (rootfs) set to be root filesystem
+	[    1.715994] 1 squashfs-split partitions found on MTD device rootfs
+	[    1.722175] 0x000001c60000-0x0000028c0000 : "rootfs_var"
+	[    1.745236] 0x0000028c0000-0x000004ec0000 : "Kernel_backup"
+	[    1.806070] 0x000004ec0000-0x0000056c0000 : "CFG"
+	[    1.822393] 0x0000056c0000-0x000005ac0000 : "RAE"
+	[    1.833183] 0x000005ac0000-0x000005bc0000 : "POT"
+	[    1.843779] 0x000005bc0000-0x000005fc0000 : "Language"
+	[    1.855067] 0x000005fc0000-0x0000061c0000 : "Traffic"
+	[    1.863392] 0x0000061c0000-0x0000062c0000 : "Cert"
+	[    1.870064] 0x0000062c0000-0x0000063c0000 : "NTGRcryptK"
+	[    1.877258] 0x0000063c0000-0x0000068c0000 : "NTGRcryptD"
+	[    1.890005] 0x0000068c0000-0x0000069c0000 : "LOG"
+	[    1.902786] mtdoops: ready 0, 1 (no erase)
+	[    1.906880] mtdoops: Attached to MTD device 19
+	[    1.911668] 0x0000069c0000-0x000007000000 : "User_data"
+	*/
+			partition@0 {
+				label = "Preloader";
+				reg = <0x00000 0x0080000>;
+				read-only;
+			};
+
+			partition@80000 {
+				label = "ATF";
+				reg = <0x80000 0x0040000>;
+				read-only;
+			};
+
+			partition@c0000 {
+				label = "Bootloader";
+				reg = <0xc0000 0x0080000>;
+				read-only;
+			};
+
+			partition@140000 {
+				label = "Config";
+				reg = <0x140000 0x0080000>;
+			};
+
+			factory: partition@1c0000 {
+				compatible = "nvmem-cells";
+				label = "Factory";
+				reg = <0x1c0000 0x0100000>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+				read-only;
+
+				macaddr_factory_7fff4: macaddr@7fff4 {
+					reg = <0x7fff4 0x6>;
+				};
+
+				macaddr_factory_7fffa: macaddr@7fffa {
+					reg = <0x7fffa 0x6>;
+				};
+			};
+
+			partition@2c0000 {
+				label = "firmware";
+				reg = <0x2c0000 0x2600000>;
+
+				compatible = "fixed-partitions";
+				#address-cells = <1>;
+				#size-cells = <1>;
+
+				partition@0 {
+					label = "kernel";
+					reg = <0x0 0x600000>;
+				};
+
+				partition@600000 {
+					label = "ubi";
+					reg = <0x600000 0x2000000>;
+				};
+			};
+
+			partition@28c0000 {
+				label = "firmware_backup";
+				reg = <0x28c0000 0x2600000>;
+				read-only;
+			};
```

Comment:
I would like to merge firmware + firmware_backup to `kernel` + `ubi`

Reply:
_To notice_: In doing so the backup kernel that gets check as the first thing in u-boot at address 0x28c0000 will be overwritten. 

OpenWrt does work and I can flash back the stock firmware using `nmrpflash` successfully.
It's just that if you're flashing back to stock and the 'normal' kernel get's damaged, it will not be able to recover itself from the backup kernel. Meaning the device is not in exact factory conditions. Recover with `nmrpflash` will work of course. I have no idea who relevant all that is.

Reply:
I just checked the boot log again after updating from OpenWrt using `*-sysupgrade.bin` and realise in it's current form it's not a good idea.

If the bootloader doesn't find the backup kernel, it writes the primary kernel from 0x2c0000 size 0x2600000, to the backup kernel location at 0x28c0000 (again with size 0x2600000).

That only happens when upgrading using the `*-sysupgrade.bin` file.

Using `nmrpflash` with the `*-factory.img` works somehow. U-boot check for backup kernel and find it -  a OpenWrt kernel - and then loads the primary kernel.